dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -2853,6 +2853,12 @@ Spdp::SpdpTransport::choose_recv_socket(ACE_HANDLE h) const
   return unicast_socket_;
 }
 
+bool valid_size(const ACE_INET_Addr& a)
+{
+  const int max_size = 16;
```

Comment:
Why is this 16?

Reply:
I tried separate max (4 bytes for IPv4, 16 bytes for IPv6) based on ACE_HAS_IPV6 and get_type(). Somehow it didn't work. Simply using 16 bytes here worked. I tested this with the RtpsDiscovery test.

Reply:
There are two separate things to consider here:
1. We don't want to have a "magic number" in the code without any explanation for why this is the magic number
2. This magic number won't work IPv6 is larger, and it's bound to be platform dependent so hard to know if a single number could ever work for all platforms.

Take a look at what ACE is doing in INET_Addr, specifically in methods like `ACE_INET_Addr::set_address`.  The `get_size()` value is computed based on `sizeof` operators.  The same approach is needed here.

Reply:
1. ACE_INET_Addr::set_address() sets addr_size_ to 4 bytes for ip4 and 16 bytes for ip6.

2. ACE_Addr::get_size() does not use sizeof operator:
ACE_Addr::get_size (void) const
{
  return this->addr_size_;
}

3. Therefore, a checking function like this should work across platforms:
bool valid_size(const ACE_INET_Addr& a)
{
  const int in6_size = 16;
  return a.get_size() <= in6_size;
}

4. The function that uses sizeof operator is get_addr_size():
ACE_INLINE int
ACE_INET_Addr::get_addr_size (void) const
{
  ACE_TRACE ("ACE_INET_Addr::get_addr_size");
#if defined (ACE_HAS_IPV6)
  if (this->get_type () == PF_INET)
    return sizeof this->inet_addr_.in4_;
  else
    return sizeof this->inet_addr_.in6_;
#else
  return sizeof this->inet_addr_.in4_;
#endif /* ACE_HAS_IPV6 */
}

5. I agree with Tim's comment that a sanity check should be in ACE object, and I would prefer the good c++ practice, i.e. constructor puts an object in valid state. I also found a lot of design issues in ACE_Addr and its favorite child ACE_INET_Addr. We have 4 options (from ideal to practical): 1) Replace ACE with std; 2) Improve ACE; 3) Wrap ACE to remove the dependency on ACE from the majority of our code; and 4) Make patches like this or the LogAddr I did before.